src/widgetastic_patternfly/__init__.py

```diff
@@ -1849,3 +1850,172 @@ def click_location(self, name, handle_alert=True):
     def read(self):
         """Return the active location of the breadcrumb"""
         return self.active_location
+
+
+class DatePicker(View):
+    """Represents the Bootstrap DatePicker.
+
+      Args:
+        name: name of DatePicker
+        id: id of DatePicker
+        locator: If none of above apply, you can also supply a full locator.
+
+    .. code-block:: python
+
+       date = DatePicker(name='miq_date_1')
+    """
+    textbox = TextInput(locator=Parameter('@locator'))
+    DATE_MAPPING = {'dd': '%d', 'mm': '%m', 'MM': '%B', 'yy': '%y', 'yyyy': '%Y'}
+
+    def __init__(self, parent, id=None, name=None, locator=None, logger=None):
+        View.__init__(self, parent=parent, logger=logger)
+        if id is not None:
+            self.locator = './/*[normalize-space(@id)={}]'.format(quote(id))
+        elif name is not None:
+            self.locator = './/*[normalize-space(@name)={}]'.format(quote(name))
+        elif locator is not None:
+            self.locator = locator
+        else:
+            raise TypeError('You need to specify either, id, name or locator for DatePicker')
+
+    class HeaderView(View):
+        prev_button = Text(".//*[contains(@class, 'prev')]")
+        next_button = Text(".//*[contains(@class, 'next')]")
+        dp_switch = Text(".//*[contains(@class, 'datepicker-switch')]")
+        _elements = {}
+
+        def select(self, value):
+            for el, web_el in self._elements.items():
+                if el == value:
+                    web_el.click()
+                    return True
+
+        @property
+        def active(self):
+            for el, web_el in self._elements.items():
+                if bool(self.browser.classes(web_el) & {'active', 'focused'}):
+                    return el
+
+    @View.nested
+    class date_pick(HeaderView):    # noqa
+        DATES = ".//*[contains(@class, 'datepicker-days')]/table/tbody/tr/td"
+        _dates = {}
+
+        @property
+        def _elements(self):
+            for el in self.browser.elements(self.DATES):
+                if not bool({'old', 'new', 'disabled'} & self.browser.classes(el)):
+                    self._dates.update({int(el.text): el})
```

Comment:
for `date_pick`, `month_pick` and `year_pick`, `select` and `active` methods are common. date and year values are `int` and others `str`. Also, I required elements for a `click` operation. I created `dict` at `_element` level only to avoid repetitive methods also to avoid repetitive iteration over elements.